src/springsettings.js

```diff
@@ -7,58 +7,81 @@ const os = require('os');
 const { writePath } = require('./spring_platform');
 
 class Springsettings extends EventEmitter {
-	applyDefaultsOnSettings(defaults, springsettingsPath) {
+	// Parsing according to implementation in rts/System/Config/ConfigSource.cpp
+	readSettings(springsettingsPath) {
 		var fileContent = '';
 		try {
 			fileContent = fs.readFileSync(springsettingsPath).toString();
 		} catch (err) {
 			// ignore errors
 		}
 		const lines = fileContent.split(/\r?\n/g);
-
-		var newContent = '';
-		var keysSeen = {};
-		// read existing settings for any different values
+		if (lines[lines.length-1] === '') lines.pop();
+		const values = {};
+		const comments = {};
+		let comment = '';
 		for (let i = 0; i < lines.length; i++) {
 			const line = lines[i];
-
-			const keyvalue = line.split(/=/);
-			if (keyvalue.length != 2) {
-				if (line.trim() != '') {
-					newContent += line + os.EOL;
-				}
+			if (line.trim() === '' || line.trim()[0] === '#') {
+				comment += line + '\n';
 				continue;
 			}
+			const keyvalue = line.split(/=/, 2);
+			if (keyvalue.length != 2) {
+				throw new Error(
+					`Error loading ${springsettingsPath}: Cannot parse line ` +
+					`${i+1} ("${line}"): it's not a comment or option assignment`);
+			}
```

Comment:
I disagree that silently eating lines that are neither correct values, not comments, is a better user experience. If something is unambiguous, user should be asked what to do about it, not program guessing the purpose, as there will be cases where it will guess wrong. Especially in thing like a config file that should be unambiguous.

I don't believe that robustness principle applies *here*, it makes sense only in some very specific cases, not in the generic sense everywhere. History showed that it has plenty of disadvantages to just apply it everywhere as a dogma:
- https://queue.acm.org/detail.cfm?id=1999945
- https://www.ietf.org/archive/id/draft-iab-protocol-maintenance-05.html

Reply:
You make reasonable points. I suppose it's fine - the application won't crash, it will print out an error (which can indeed be better than silently ignoring incorrect configs, seeing as incorrect configs are likely to arise from user attempts at modifying configs manually).